### ISSUE TITLE
[Bug]: FabricObject.prototype.set  does not take effect

### ISSUE THREAD

**Issue by Yi-Wo-Zuo** [fabricjs/fabric.js#10031]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

6.0.2

### Node Version (if applicable)

None

### Link To Reproduction

none

### Steps To Reproduce

1. npm install fabric
2. import { FabricObject } from "fabric";
3. FabricObject.prototype.set({
  cornerStrokeColor: "#000000", 
  cornerColor: "#000000", 
  cornerStyle: "square",
  transparentCorners: false,
  cornerSize: 6, //边角大小
  borderScaleFactor: 1, 
  borderColor: "#000000", 
  borderDashArray: [6],
  padding: 5,
});

### Expected Behavior

import { FabricObject } from "fabric";

FabricObject.prototype.set({
  cornerStrokeColor: "#000000", 
  cornerColor: "#000000", 
  cornerStyle: "square",
  transparentCorners: false,
  cornerSize: 6, //边角大小
  borderScaleFactor: 1, 
  borderColor: "#000000", 
  borderDashArray: [6],
  padding: 5,
});

### Actual Behavior

import { FabricObject } from "fabric";

FabricObject.prototype.set({
  cornerStrokeColor: "#000000", 
  cornerColor: "#000000", 
  cornerStyle: "square",
  transparentCorners: false,
  cornerSize: 6, //边角大小
  borderScaleFactor: 1, 
  borderColor: "#000000", 
  borderDashArray: [6],
  padding: 5,
});

**Comment by zhe-he** [comment]
Hmm... Do you want to set a default value? Or intercept the set? Use method a for setting a default value and method b for intercepting.

```typescript
import { FabricObject } from "fabric";
// a
FabricObject.ownDefaults.borderColor = "#000000"

// b
const lastSet = FabricObject.prototype.set;
FabricObject.prototype.set = function(k, v) {
    console.log("before set", k, v);

    const r = lastSet.call(this, k, v);

    console.log("after set", k, v);

    return r;
}

```

**Comment by asturur** [comment]
You can't use the method set directly from the prototype in that way, you are calling it without context it doesn't have a THIS that make sense.

**Comment by folknor** [comment]
Is it impossible to set custom props now? I was unsure if this was related to this issue or not.

I have for example
```js
loadSVGFromURL(icon.svg, (_, svg) => {
	var size = util.getRandomInt(24, 48)
	svg.scaleToWidth(size)
	svg.set({
		left: util.getRandomInt(size / 2, maxWidth - size),
		top: util.getRandomInt(size / 2, maxHeight - size),
		id: icon.id,
		origSize: size,
		lockRotation: true,
		lockScalingX: true,
		lockScalingY: true,
		hasControls: false
	})
	svg.setCoords()
	canvas.add(svg)
})
```

But `id: icon.id,` this property is at least not serialized when I do for example `canvas.toObject(["id", "origSize", "centered"])`

This worked in 5.x.x before

**Comment by asturur** [comment]
it should be serialized, if it doesn't is a bug.
Can you make a reproducible example?

**Comment by folknor** [comment]
When I made a reduced test case it worked fine lol

So no bug, my fault gg

**Comment by asturur** [comment]
Yeah that is usually the outcome. When you focus on reproducing in an isolated case you fix it.

**Comment by DreamTear** [comment]
you need do like this:
```javascript
Object.assign(FabricObject.ownDefaults, {
            cornerColor: '#fff',
            cornerStyle: 'circle',
            borderColor: '#3b82f6',
            borderScaleFactor: 1.5,
            transparentCorners: false,
            borderOpacityWhenMoving: 1,
            cornerStrokeColor: '#3b82f6',
        })
```
it works
